main is red: `RunCommandFailFastAsync` used but undefined in TypeScriptEmptyAppHostTemplateTests (introduced by #17575)

## Summary

`main` does not compile its CLI E2E test project, so `ci.yml` is failing on every PR (and on `main` itself). This blocks all PR merges until fixed.

## Root cause

Commit `4cfc99df7` (PR #17575, "Fix TypeScript AppHost async callback deadlock") added `tests/Aspire.Cli.EndToEnd.Tests/TypeScriptEmptyAppHostTemplateTests.cs`, which calls a helper that does not exist:

```
tests/Aspire.Cli.EndToEnd.Tests/TypeScriptEmptyAppHostTemplateTests.cs(79): await auto.RunCommandFailFastAsync("cd TsDeadlockRepro", counter);
tests/Aspire.Cli.EndToEnd.Tests/TypeScriptEmptyAppHostTemplateTests.cs(80): await auto.RunCommandFailFastAsync("aspire restore --non-interactive", counter, TimeSpan.FromMinutes(3));
tests/Aspire.Cli.EndToEnd.Tests/TypeScriptEmptyAppHostTemplateTests.cs(81): await auto.RunCommandFailFastAsync("npm run build", counter, TimeSpan.FromMinutes(2));
```

`Hex1bTerminalAutomator` (from `Hex1b.Automation`) exposes `RunCommandAsync`, which is used everywhere else in the project, but there is **no** `RunCommandFailFastAsync` defined anywhere in the repo or the referenced package version. Build fails with:

```
error CS1061: 'Hex1bTerminalAutomator' does not contain a definition for 'RunCommandFailFastAsync'
```

## Evidence

- `ci.yml` is `failure` on `main` HEAD `4cfc99df7` (and the prior commit), failing checks **Stabilization Check** and **Tests / Setup for tests**.
- `git grep RunCommandFailFastAsync origin/main` returns only the three call sites above — no definition.

## Suggested fix

Either add the missing `RunCommandFailFastAsync` helper (a fail-fast variant of `RunCommandAsync` that asserts non-zero exit fails the test), bump the Hex1b package to a version that provides it, or replace the calls with the existing `RunCommandAsync`. Alternatively revert #17575 until the helper is available.

Confirmed this is now blocking PR CI repo-wide (not just main's own build). Example: PR #17680's `Tests / Setup for tests` job fails to compile with:

```
tests/Aspire.Cli.EndToEnd.Tests/TypeScriptEmptyAppHostTemplateTests.cs(79,24): error CS1061: 'Hex1bTerminalAutomator' does not contain a definition for 'RunCommandFailFastAsync'
```

On `origin/main` only `RunCommandAsync` exists (`tests/Shared/Hex1bAutomatorTestHelpers.cs:265`); there is no `RunCommandFailFastAsync` definition anywhere, and no open fix PR. Every PR whose CI merges with main is blocked until the missing helper is added (or #17575's test changes are reverted). cc the #17575 author to add the intended `RunCommandFailFastAsync` helper.

<details><summary>🤖 Aspire triage pass — 2026-05-29 21:24 UTC</summary>

**Area**: `area-engineering-systems` (already applied)
**Kind**: test-failure (build break in CI; blocks PR merges)
**Repro**: yes-minimal — `dotnet build tests/Aspire.Cli.EndToEnd.Tests/` reproduces `error CS1061` on `RunCommandFailFastAsync`; details in the issue body and the comment from the reporter
**Affected version(s)**: `main` HEAD (post-#17575 merge on 2026-05-29); not yet in a released bit
**Severity**: S2 — repo-wide PR CI failing (`Tests / Setup for tests`, `Stabilization Check`), no workaround other than reverting / patching `main`

**Urgency**: 🟡 vNext — recommend milestone **13.5**
Reason: CI break on `main` introduced 2026-05-29 by #17575 (test-only file references a `Hex1bTerminalAutomator.RunCommandFailFastAsync` helper that does not exist in the repo or the referenced `Hex1b.Automation` version). Not a shipped-product regression in 13.4 (the failure is in the test project, not the 13.4 product surface), so it does not clear the 13.4 release-blocking bar as defined for this snap. However it is *operationally* urgent because it blocks landing any other 13.4 fixes in the final week before ship — flagging for human escalation so a release manager can decide whether to pull into 13.4 servicing (or just hotfix `main` and the release branch in parallel).

**Suggested owner(s)**: @IEvangelist (author of the regressing PR #17575, score 5), backup @radical (score 9 on `area-engineering-systems` general activity)
Evidence:
 - #17575 ("Fix TypeScript AppHost async callback deadlock") merged by @IEvangelist 2026-05-29T19:24Z added `tests/Aspire.Cli.EndToEnd.Tests/TypeScriptEmptyAppHostTemplateTests.cs` containing the three undefined calls — direct authorship of the break
 - @radical has the highest recency-weighted merged-PR count in `area-engineering-systems` over the last 90 days (e.g. #16478, #16454, #16446, #15127, #15111, #15110, #15055, #15031, #15023, #15016) — natural backup owner for the test-infra side
 - CODEOWNERS not separately consulted — the regressing PR author is unambiguous here

**Confidence**: high (root cause + introducing commit are explicitly identified by the reporter and confirmed by the comment evidence; my own PR lookup confirms #17575 added the file 2026-05-29)
**Needs human?**: yes — repo-wide CI break in the final week before 13.4 ships; release manager needs to decide between (a) revert #17575, (b) add the missing `RunCommandFailFastAsync` helper, (c) swap calls to existing `RunCommandAsync`, or (d) bump `Hex1b.Automation`

_Dry-run mode — no labels or milestone applied (beyond `triage:bot-seen`). Reply `/triage apply` to promote, or edit and reapply._
</details>

--- a/tests/Aspire.Cli.EndToEnd.Tests/TypeScriptEmptyAppHostTemplateTests.cs
+++ b/tests/Aspire.Cli.EndToEnd.Tests/TypeScriptEmptyAppHostTemplateTests.cs
@@ -59,57 +59,24 @@ public sealed class TypeScriptEmptyAppHostTemplateTests(ITestOutputHelper output
         var workspace = TemporaryWorkspace.Create(output);
 
         using var terminal = CliE2ETestHelpers.CreateDockerTestTerminal(repoRoot, strategy, output, mountDockerSocket: true, workspace: workspace);
-
-        var pendingRun = terminal.RunAsync(TestContext.Current.CancellationToken);
-
         var counter = new SequenceCounter();
         var auto = new Hex1bTerminalAutomator(terminal, defaultTimeout: TimeSpan.FromSeconds(500));
-        var testBodyFailed = false;
+        await using var terminalRun = CliE2ETestHelpers.StartRun(terminal, workspace, auto, counter, output, TestContext.Current.CancellationToken);
 
-        try
-        {
-            await auto.PrepareDockerEnvironmentAsync(counter, workspace, enableDcpDiagnostics: true);
-            await auto.InstallAspireCliAsync(strategy, counter);
+        await auto.PrepareDockerEnvironmentAsync(counter, workspace, enableDcpDiagnostics: true);
+        await auto.InstallAspireCliAsync(strategy, counter);
 
-            await auto.AspireNewTypeScriptEmptyAppHostAsync("TsDeadlockRepro", counter);
+        await auto.AspireNewTypeScriptEmptyAppHostAsync("TsDeadlockRepro", counter);
 
-            var appDirectory = Path.Combine(workspace.WorkspaceRoot.FullName, "TsDeadlockRepro");
-            WriteDeadlockReproFiles(appDirectory);
+        var appDirectory = Path.Combine(workspace.WorkspaceRoot.FullName, "TsDeadlockRepro");
+        WriteDeadlockReproFiles(appDirectory);
 
-            await auto.RunCommandFailFastAsync("cd TsDeadlockRepro", counter);
-            await auto.RunCommandFailFastAsync("aspire restore --non-interactive", counter, TimeSpan.FromMinutes(3));
-            await auto.RunCommandFailFastAsync("npm run build", counter, TimeSpan.FromMinutes(2));
+        await auto.RunCommandAsync("cd TsDeadlockRepro", counter);
+        await auto.RunCommandAsync("aspire restore --non-interactive", counter, TimeSpan.FromMinutes(3));
+        await auto.RunCommandAsync("npm run build", counter, TimeSpan.FromMinutes(2));
 
-            await auto.AspireStartAsync(counter, startTimeout: TimeSpan.FromMinutes(2));
-            await auto.AspireStopAsync(counter);
-        }
-        catch
-        {
-            testBodyFailed = true;
-            throw;
-        }
-        finally
-        {
-            try
-            {
-                await auto.CaptureAspireDiagnosticsAsync(counter, workspace);
-            }
-            catch { } // Best effort
-
-            try
-            {
-                await auto.TypeAsync("exit");
-                await auto.EnterAsync();
-                await pendingRun;
-            }
-            catch
-            {
-                if (!testBodyFailed)
-                {
-                    throw;
-                }
-            }
-        }
+        await auto.AspireStartAsync(counter, startTimeout: TimeSpan.FromMinutes(2));
+        await auto.AspireStopAsync(counter);
     }
 
     private static void WriteDeadlockReproFiles(string appDirectory)

